classes/ETL/Ingestor/HostTableTransformIngestor.php

```diff
@@ -0,0 +1,105 @@
+<?php
+
+namespace ETL\Ingestor;
+
+use ETL\aOptions;
+use ETL\iAction;
+use ETL\aAction;
+use ETL\Configuration\EtlConfiguration;
+use ETL\EtlOverseerOptions;
+
+use Log;
+
+class HostTableTransformIngestor extends pdoIngestor implements iAction
+{
+
+    private $_instance_state;
+    private $_end_time;
+
+    /**
+     * @see ETL\Ingestor\pdoIngestor::__construct()
+     */
+    public function __construct(aOptions $options, EtlConfiguration $etlConfig, Log $logger = null)
+    {
+        parent::__construct($options, $etlConfig, $logger);
+
+        $this->_end_time = $etlConfig->getVariableStore()->endDate ? date('Y-m-d H:i:s', strtotime($etlConfig->getVariableStore()->endDate)) : null;
+
+        $this->resetInstance();
+    }
+
+    private function initInstance($srcRecord)
+    {
+        $default_end_time = isset($this->_end_time) ? $this->_end_time : $srcRecord['record_time'];
+
+        $this->_instance_state = array(
+            'host_id' => $srcRecord['host_id'],
+            'node_id' => $srcRecord['node_id'],
+            'start_time' => $srcRecord['record_time'],
+            'start_day_id' => $srcRecord['record_day_id'],
+            'end_time' => $srcRecord['record_time'],
+            'end_day_id' => $srcRecord['record_day_id'],
+        );
+    }
+
+    private function resetInstance()
+    {
+        $this->_instance_state = null;
+    }
+
+    private function updateInstance($srcRecord)
```

Comment:
If `$srcRecord` is an array, use type hinting to ensure this. For example, `private function updateInstance(array $srcRecord)`